libr/asm/p/asm_x86_nz.c

```diff
@@ -1254,13 +1267,12 @@ SETNP/SETPO - Set if No Parity / Set if Parity Odd (386+)
 					return 0;
 				} else {
 					// mov [num], reg
-					ut8 *ptr = (ut8 *)&num;
 					data[l++] = 0x89;
 					data[l++] = (r0 << 3) | 5;
-					data[l++] = ptr[0];
-					data[l++] = ptr[1];
-					data[l++] = ptr[2];
-					data[l++] = ptr[3];
+					data[l++] = num;
```

Comment:
for `mov [num], reg` if `num` is negative the opcode is fubar anyway compared to `nasm` output, and needs to be fixed. For positive values of `num` the output is fine. I'll be carrying on with this anyway.